signal/signals.c

```diff
@@ -198,18 +198,29 @@ bool StatusServiceSignalEvent(const char *jsonEvent) {
 
 #else
 // ======================================================================================
-// cgo compilation (for local tests)
+// cgo compilation (for desktop platforms and local tests)
 // ======================================================================================
 
 #include <stdio.h>
 #include <stddef.h>
 #include <stdbool.h>
 #include "_cgo_export.h"
 
+typedef void (*callback)(const char *jsonEvent);
+callback gCallback = 0;
+
 bool StatusServiceSignalEvent(const char *jsonEvent) {
-    NotifyNode((char *)jsonEvent); // re-send notification back to status node
+	if (gCallback) {
+		gCallback(jsonEvent);
+	} else {
+		NotifyNode((char *)jsonEvent); // re-send notification back to status node
+	}
 
 	return true;
 }
 
+void SetJailSignalCb(void* cb) {
```

Comment:
`s/void* cb/void *cb` to keep consistency.

Also, this is not related to jail. Signals are sent from different parts of the app. `SetSignalEventCallback(void *cb)` sounds more appropriate.